### PR TITLE
feat: swap blst for napi library

### PR DESCRIPTION
**Pull request by matthewkeil**
**Motivation**

DRAFT PR.  Not ready for full review.

**NOTE: most of the diff is the blst-ts (with the full supranational/bls) library itself.  It has not been published so all that code is brought into lodestar to test the code**

Branch is deployed to `feat2` to collect metrics

**Description**

Swaps all references to `@chainsafe/bls` and `@chainsafe/blst` with the new library `@chainsafe/blst-ts`

### CODE REVIEW

File: packages/beacon-node/src/chain/bls/multithread.ts [hunk 1]
```diff
@@ -348,34 +372,64 @@ export class BlsMultiThreadWorkerPool implements IBlsVerifier {
...
+    // Prepare work package
+    const prepared = this.prepareWork();
+    if (prepared.length === 0) {
+      return;
+    }
+
+    if (this.blsPoolType === BlsPoolType.workers) {
+      await this.runJobWorkerPool(prepared);
+    } else {
+      await this.runJobLibuv(prepared);
```
Comment (twoeths) [inline packages/beacon-node/src/chain/bls/multithread.ts:412]: if possible I'd like to refactor the current implementation to have `BaseBlsMultiThreadPool` to move all common code there, have `BlsMultiThreadWorkerPool` for worker specific implementation, and have this new work in another implementation, so there would be multiple steps required to make this happens similar to `n_historical_states` work I've worked on

that would make a smooth transition to new bls implementation, it's also easier to review and maintain. What's your opinions on this approach @matthewkeil @wemeetagain ?

**Comment by twoeths** [comment]
the test mainnet node does not look great in the last 6h, may worth to look into this @matthewkeil 

Screenshot 2024-01-26 at 20 20 20

vs unstable

Screenshot 2024-01-26 at 20 21 57

**Comment by matthewkeil** [comment]
Closing this in favor of #6362

File: packages/beacon-node/src/chain/bls/multithread.ts (hunk 1 above)
Comment (matthewkeil) [reply packages/beacon-node/src/chain/bls/multithread.ts:412]: Sounds good.  I can take a look at this in the updated PR #6362 once the metrics look good.  Going to try and limit the complexity until I can get it running smoothly though @tuyennhv

**Pull request closed without merging**